core/dataset.h

```diff
@@ -685,6 +685,8 @@ struct Dataset: public MldbEntity {
        This will return the name that the row has in the table with this alias*/
     virtual RowName getOriginalRowName(const Utf8String& tableName,
                                        const RowName & name) const;
+
+    virtual size_t getRowCount() const;
```

Comment:
Minor nitpick: should be uint64_t, since on 32 bits an external dataset may have more than 4 billion rows.